File: app/src/main/java/com/snaptiongame/app/data/services/notifications/IdentificationService.java
package com.snaptiongame.app.data.services.notifications;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;
import com.snaptiongame.app.data.auth.AuthManager;
import com.snaptiongame.app.data.providers.NotificationProvider;

/**
 * @author Tyler Wong
 */

public class IdentificationService extends FirebaseInstanceIdService {
    @Override
    public void onTokenRefresh() {
        super.onTokenRefresh();

        if (AuthManager.isLoggedIn()) {
            sendRegistrationToServer(FirebaseInstanceId.getInstance().getToken());
        }
    }

    private void sendRegistrationToServer(String token) {
        NotificationProvider.refreshNotificationToken(token);
    }
}

File: app/src/main/java/com/snaptiongame/app/data/services/notifications/IdentificationService.kt
package com.snaptiongame.app.data.services.notifications

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import com.snaptiongame.app.data.auth.AuthManager
import com.snaptiongame.app.data.providers.*

/**
 * @author Tyler Wong
 */

class IdentificationService : FirebaseInstanceIdService() {
    override fun onTokenRefresh() {
        super.onTokenRefresh()

        if (AuthManager.isLoggedIn()) {
            sendRegistrationToServer(FirebaseInstanceId.getInstance().token)
        }
    }

    private fun sendRegistrationToServer(token: String?) {
        refreshNotificationToken(token ?: "")
    }
}